source/math/IntVectorList.ooc

```diff
@@ -0,0 +1,68 @@
+/*
+* Copyright (C) 2014 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+use ooc-collections
+import math
+import FloatVectorList
+
+IntVectorList: class extends VectorList<Int> {
+	init: func ~default {
+		this super()
+	}
+	init: func ~heap (capacity: Int) {
+		super(capacity)
+	}
+	init: func ~fromVectorList (other: VectorList<Int>) {
+		this super(other _vector)
+		this _count = other count
+	}
+	operator [] <T> (index: Int) -> T {
+		this as VectorList<Int> _vector[index]
+	}
+	operator []= (index: Int, item: Int) {
+		this _vector[index] = item
+	}
+	sort: func {
+		inOrder := false
+		while (!inOrder) {
+			inOrder = true
+			for (i in 0 .. this count - 1)
+				if (this[i] > this[i + 1]) {
+					inOrder = false
+					tmp := this[i]
+					this[i] = this[i + 1]
+					this[i + 1] = tmp
+				}
+		}
+	}
+	compress: func -> This {
```

Comment:
> getSet

Yeah, not clear enough, unless you are already in a set theory mindset.

> getUniqueElements or getUniques, maybe?

`compress` is actually not bad, if you are familiar with the term, but this suggestion works for people who aren't, too. If you want to keep the name, though, you should at least leave a comment explaining what it does - `[1,1,2,2,5,6,7,7,7] -> [1,2,5,6,7]` is very pedagogical, I think, but perhaps
```ooc
// Returns a new list without duplicate elements, e.g. [1,1,2,2,5,6,7,7,7] -> [1,2,5,6,7]
```
Still, if you call it `getUniqueElements` (or `uniqueElements`, I'm not sure where we draw the line for a property to be honest), you won't need a comment. Up to @marcusnaslund .